homeassistant/components/weatherflow_cloud/sensor.py

```diff
@@ -176,35 +302,161 @@ async def async_setup_entry(
 
-    coordinator: WeatherFlowCloudDataUpdateCoordinator = hass.data[DOMAIN][
-        entry.entry_id
-    ]
+    coordinators: WeatherFlowCoordinators = hass.data[DOMAIN][entry.entry_id]
+    rest_coordinator = coordinators.rest
+    wind_coordinator: WeatherFlowCloudDataCallbackCoordinator[
+        EventDataRapidWind, EventDataRapidWind, RapidWindListenStartMessage
+    ] = coordinators.wind
+    observation_coordinator: WeatherFlowCloudDataCallbackCoordinator[
+        WebsocketObservation, WebsocketObservation, ListenStartMessage
+    ] = coordinators.observation
 
-    async_add_entities(
-        WeatherFlowCloudSensor(coordinator, sensor_description, station_id)
-        for station_id in coordinator.data
+    entities: list[SensorEntity] = [
+        WeatherFlowCloudSensorREST(rest_coordinator, sensor_description, station_id)
+        for station_id in rest_coordinator.data
         for sensor_description in WF_SENSORS
+    ]
+
+    entities.extend(
+        WeatherFlowWebsocketSensorWind(
+            coordinator=wind_coordinator,
+            description=sensor_description,
+            station_id=station_id,
+            device_id=device_id,
+        )
+        for station_id in wind_coordinator.stations.station_outdoor_device_map
+        for device_id in wind_coordinator.stations.station_outdoor_device_map[
+            station_id
+        ]
+        for sensor_description in WEBSOCKET_WIND_SENSORS
     )
 
+    entities.extend(
+        WeatherFlowWebsocketSensorObservation(
+            coordinator=observation_coordinator,
+            description=sensor_description,
+            station_id=station_id,
+            device_id=device_id,
+        )
+        for station_id in observation_coordinator.stations.station_outdoor_device_map
+        for device_id in observation_coordinator.stations.station_outdoor_device_map[
+            station_id
+        ]
+        for sensor_description in WEBSOCKET_OBSERVATION_SENSORS
+    )
+    async_add_entities(entities)
 
-class WeatherFlowCloudSensor(WeatherFlowCloudEntity, SensorEntity):
-    """Implementation of a WeatherFlow sensor."""
 
-    entity_description: WeatherFlowCloudSensorEntityDescription
+class WeatherFlowSensorBase(WeatherFlowCloudEntity, SensorEntity, ABC):
+    """Common base class."""
 
     def __init__(
         self,
-        coordinator: WeatherFlowCloudDataUpdateCoordinator,
-        description: WeatherFlowCloudSensorEntityDescription,
+        coordinator: WeatherFlowCloudUpdateCoordinatorREST
+        | WeatherFlowCloudDataCallbackCoordinator[
+            EventDataRapidWind | WebsocketObservation,
+            RapidWindListenStartMessage | ListenStartMessage,
+            RapidWindWS | WebsocketObservation,
+        ],
+        description: WeatherFlowCloudSensorEntityDescription
+        | WeatherFlowCloudSensorEntityDescriptionWebsocketWind
+        | WeatherFlowCloudSensorEntityDescriptionWebsocketObservation,
```

Comment:
I kinda really dislike this typing

Reply:
I made type aliases - not sure if that fixes anything but it makes it more readable I think